IdentityCore/src/MSIDJsonSerializableFactory.m

```diff
@@ -21,52 +21,50 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-#import "MSIDBrokerOperationRequestFactory.h"
-#import "MSIDBrokerOperationRequest.h"
+#import "MSIDJsonSerializableFactory.h"
 #import "MSIDJsonSerializable.h"
-#import "NSDictionary+MSIDExtensions.h"
-#import "MSIDConstants.h"
 
-static NSMutableDictionary *s_operationRequestClasses = nil;
+static NSMutableDictionary<NSString *, Class<MSIDJsonSerializable>> *s_container = nil;
 
-@implementation MSIDBrokerOperationRequestFactory
+@implementation MSIDJsonSerializableFactory
 
-+ (void)registerOperationRequestClass:(Class<MSIDJsonSerializable>)operationRequestClass
-                            operation:(NSString *)operation
++ (void)registertClass:(Class<MSIDJsonSerializable>)class forKey:(NSString *)key
 {
-    if (!operationRequestClass || !operation) return;
-    if ([operationRequestClass isKindOfClass:MSIDBrokerOperationRequest.class]) return;
+    if (!class || !key) return;
+    if (![key isKindOfClass:NSString.class]) return;
+    if (![class conformsToProtocol:@protocol(MSIDJsonSerializable)]) return;
     
     @synchronized(self)
     {
         static dispatch_once_t once;
         dispatch_once(&once, ^{
-            s_operationRequestClasses = [NSMutableDictionary new];
+            s_container = [NSMutableDictionary new];
```

Comment:
I had some additional thoughts about this approach, as it seems like we're using it in many classes now.
I was wondering if we could run a quick test with and without class registration on +load (with e.g. 10-20-100 classes) and see how much time on app startup we're loosing because of using this approach? I'm a bit worried that there's a lot happening here, and this is all pre-main() time, so it might impact app startup performance. However, on this device there might be even no SSO extension present at all. See context here: https://developer.apple.com/videos/play/wwdc2016/406/

Reply:
I did some tests and I didn't find any performance issue there.

Reply:
Agreed to keep current logic.